src/HeadlessUniversalCheckout/Managers/VaultManager.ts

```diff
@@ -17,22 +17,23 @@ class PrimerHeadlessUniversalCheckoutVaultManager {
     ///////////////////////////////////////////
 
     async configure(): Promise<void> {
-            return new Promise(async (resolve, reject) => {
-                try {
-                    await RNPrimerHeadlessUniversalCheckoutVaultManager.configure();
-                    resolve();
-                } catch (err) {
-                    console.error(err);
-                    reject(err);
-                }
-            });
-        }
+        return new Promise(async (resolve, reject) => {
+            try {
+                await RNPrimerHeadlessUniversalCheckoutVaultManager.configure();
+                resolve();
+            } catch (err) {
+                console.error(err);
+                reject(err);
+            }
+        });
+    }
 
-    async fetchVaultedPaymentMethods(): Promise<PrimerVaultedPaymentMethod[]> {
+    async fetchVaultedPaymentMethods(): Promise<VaultedPaymentMethod[]> {
         return new Promise(async (resolve, reject) => {
             try {
-                const paymentMethods: PrimerVaultedPaymentMethod[] =
+                const data =
```

Comment:
Is there a reason to lose the type information here?

Reply:
Hey @jnewc Originally I was unwrapping the paymentMethods wrongly, I presumed that we would get a `PrimerVaultedPaymentMethod[]` array back but in actual fact we get an object with a `paymentMethods` field on it that contains the array.

Reply:
It feels like we're missing exposure of the result model then. I'd expect to see a `VaultedPaymentMethodResult` type or something like that, whether bridged from native or defined as a typescript type.

Reply:
Resolved in latest changes!